src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1019,7 +1024,11 @@ Namespace Microsoft.CodeAnalysis.Operations
             Dim constantValue As [Optional](Of Object) = New [Optional](Of Object)()
             Dim isImplicit As Boolean = boundCaseBlock.WasCompilerGenerated
 
-            Dim condition As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() Create(boundCaseBlock.CaseStatement.ConditionOpt))
+            ' Some bound nodes used by the boundCaseBlock.CaseStatement.CaseClauses are also going to be used in boundCaseBlock.CaseStatement.ConditionOpt.
+            ' If we simply create another tree based on boundCaseBlock.CaseStatement.ConditionOpt, due to the caching we will end up with the same
+            ' IOperation nodes in two trees, and two parents will compete for assigning itself as the parent - trouble. To avoid that, we simply use
+            ' a new factory to create IOperation tree for the condition.
+            Dim condition As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() Clone().Create(boundCaseBlock.CaseStatement.ConditionOpt))
```

Comment:
>Clone [](start = 86, length = 5)

Are these nodes nested inside the ConditionOpt? ie it's not the condition itself, but the children of the condition?

Reply:
> Are these nodes nested inside the ConditionOpt? ie it's not the condition itself, but the children of the condition?

Yes, they are the Values of the case clauses and are also used as operands of binary operators inside the condition.

---
In reply to: [187495725](https://github.com/dotnet/roslyn/pull/26771#discussion_r187495725) [](ancestors = 187495725)

Reply:
> This sub-tree or nodes from it are not available to regular consumers, it is internal and used only by CFG builder.

Thanks for the clarification. Can you please add this to the comment, or comment on Clone about this?

Reply:
> Can you please add this to the comment, or comment on Clone about this?

Sure, I'll add this information to the comment above in the next PR. 

---
In reply to: [187683676](https://github.com/dotnet/roslyn/pull/26771#discussion_r187683676) [](ancestors = 187683676)